backend/app/user_handler.go

```diff
@@ -350,6 +407,27 @@ func (h *Handler) LoginUserHandler(c *gin.Context) {
 		return
 	}
 
+	// Check KYC verification status without blocking login
+	if h.kycClient != nil {
+		sponseeKeyPair, err := sr25519.Scheme{}.FromPhrase(user.Mnemonic, "")
+		if err == nil { // Only proceed if we can create the keypair
+			sponseeAddress, err := sponseeKeyPair.SS58Address(internal.SS58AddressFormat)
+			if err == nil { // Only proceed if we can get the address
+				if sponsored, err := h.kycClient.IsUserVerified(context.Background(), sponseeAddress, sponseeKeyPair); err == nil {
+					// Only update if verification check was successful
+					if user.Sponsored != sponsored {
+						user.Sponsored = sponsored
+						if err := h.db.UpdateUserByID(&user); err != nil {
+							log.Error().Err(err).Msg("failed to update user sponsorship status")
+						}
+					}
+				} else {
+					log.Error().Err(err).Msg("failed to check KYC verification status")
+				}
+			}
+		}
+	}
+
```

Comment:
At this point you hit the wall if there was an error on 415

Reply:
what do you mean?